backend/static/js/keywordSearch.js

```diff
@@ -0,0 +1,76 @@
+// this function removes the highlight
+function unHighlight(input_str){
+  return input_str.replaceAll(/\<span class\=\"keyword-search-highlight\"\>(.*?)\<\/span\>/gi, "$1");
+}
+
+// this function highlights the matching sections
+function highlightWords(input_str, words) {
+  let return_str = unHighlight(input_str.toLowerCase());
+  // find matching sections
+  let matching_sections = [];
+  for (word of words){
+    let startIndex = 0;
+    while ((index = return_str.indexOf(word, startIndex)) > -1){
+      matching_sections.push([index, index + word.length]);
+      startIndex = index + word.length;
+    }
+  }
```

Comment:
With `input_str.match(regex)` this code would be simpler and more powerful (one could also type regexes then, which doesn't harm)

Reply:
Thanks for the suggestion.
I wanted to revisit regex for a while now and this was a great opportunity!

While i agree that this part of the code (finding matching sections in one `example name` ) got simpler,
I feel like the other part (filtering `example names` that match all given keywords) got more complex and harder to read.
But i think it works now, and searching with regex-keywords is pretty cool.

I'm not 100% sure i understood the suggestion perfectly.
Does the new version behave as you would have expected?